trie/snapshotTrieStorageManager.go

```diff
@@ -35,17 +38,40 @@ func (stsm *snapshotTrieStorageManager) Get(key []byte) ([]byte, error) {
 		return nil, errors.ErrContextClosing
 	}
 
-	val, err := stsm.mainSnapshotStorer.GetFromOldEpochsWithoutAddingToCache(key)
+	val, epoch, err := stsm.mainSnapshotStorer.GetFromOldEpochsWithoutAddingToCache(key)
 	if errors.IsClosingError(err) {
 		return nil, err
 	}
 	if len(val) != 0 {
+		stsm.putInPreviousStorerIfAbsent(key, val, epoch)
 		return val, nil
 	}
 
 	return stsm.getFromOtherStorers(key)
 }
 
+func (stsm *snapshotTrieStorageManager) putInPreviousStorerIfAbsent(key []byte, val []byte, epoch core.OptionalUint32) {
+	if !epoch.HasValue {
+		return
+	}
+
+	if epoch.Value >= stsm.epoch-1 {
```

Comment:
so basically, if we are in current epoch 500, epoch will be 499 and we test for epoch 498?
Is this >= correct?

Reply:
The idea is that if the value was not retrieved from the current storer or previous storer, it needs to be saved to the previous storer.